### PR TITLE
Allow PlatformColor to return user-defined named asset color

### PR DESCRIPTION
**Pull request by danilobuerger**
## Summary

According to https://reactnative.dev/docs/platformcolor:

> You can use the PlatformColor function to access native colors on the target platform by supplying the native color’s corresponding string value.

This should also include named asset colors as they are part of the colors available on the iOS platform.

See 
https://developer.apple.com/documentation/uikit/uicolor/2877380-colornamed?language=objc
https://developer.apple.com/library/archive/documentation/Xcode/Reference/xcode_ref-Asset_Catalog_Format/Named_Color.html

This PR allows using `PlatformColor` to get a user-defined named asset color in the same manner that it works on android via xml resources:

Screenshot 2020-10-05 at 12 45 33

```ts
<View style={{ backgroundColor: PlatformColor("MyCustomColor") }}>...</View>
```

would set the background color of the view to the named asset color "MyCustomColor". Currently this doesn't work.

## Changelog

[iOS] [Added] - Allow PlatformColor to return user-defined named asset color

## Test Plan

1. Define an asset color as specified here: https://developer.apple.com/documentation/xcode/supporting_dark_mode_in_your_interface?language=objc

2. Access the color via `PlatformColor("asset color name")`

### CODE REVIEW

**Review by alshamiri5 — APPROVED** [review]

**Comment by danilobuerger** [comment]
Hi @shergin 👋 I would love to move this PR forward. Is there anything missing in order to get it merged?

**Comment by zachgibson** [comment]
@danilobuerger how do you supply custom Android colors?

**Comment by TheSavior** [comment]
The examples for PlatformColor in RNTester have tests for this behavior on Android. Can you add some examples to iOS to ensure that this keeps working?

File: React/Base/RCTConvert.m [hunk 1]
```diff
@@ -850,7 +850,11 @@ + (UIColor *)UIColor:(id)json
     if ((value = [dictionary objectForKey:@"semantic"])) {
       if ([value isKindOfClass:[NSString class]]) {
         NSString *semanticName = value;
-        UIColor *color = RCTColorFromSemanticColorName(semanticName);
+        UIColor *color = [UIColor colorNamed:semanticName];
```
Comment (appden) [inline React/Base/RCTConvert.m:853]: This API was officially added for iOS 11, but we still officially support iOS 10+. This should probably be guarded by `if (@available(iOS 11.0, tvOS 11.0, *))`.

**Review by appden — CHANGES_REQUESTED** [review]

File: React/Base/RCTConvert.m (hunk 1 above)
Comment (appden) [reply React/Base/RCTConvert.m:853]: @danilobuerger Following up to see if you can update this. Thanks!

Comment (danilobuerger) [reply React/Base/RCTConvert.m:853]: @appden This seems no longer neccessary with https://github.com/facebook/react-native/commit/a1d626739d95d6cbbb1be169b93952cdd1465486 ?

**Comment by danilobuerger** [comment]
@TheSavior You asked me the same thing in another PR (#29728) which I happily complied with and invested time in. Now that PR has again not seen any progress in nearly 3 months. So I wonder if I really should invest time again to add this here if my PRs just don't get merged. I started both PRs in August, now its January. It feels disrespectful to me if I invest all that time and effort into making PRs, but then they never get merged or answered in a reasonable time frame by the maintainers.

**Comment by oblador** [comment]
@danilobuerger Can you try rebasing this branch? I suspect the failing `Facebook Internal` tests are due to the iOS 10 issues.

**Pull request closed without merging**